engine/source/public/core/fileio.hpp

```diff
@@ -0,0 +1,48 @@
+﻿#pragma once
+
+#include <map>
+#include <string_view>
+#include <vector>
+
+namespace lumina
+{
+    class Engine;
+
+    class FileIO
+    {
+    public:
+        enum class Directory
+        {
+            Assets,
+            Config,
+            Log
+        };
+
+        FileIO();
+        ~FileIO() = default;
+
+        FileIO(const FileIO&)            = delete;
+        FileIO(FileIO&&)                 = delete;
+        FileIO& operator=(const FileIO&) = delete;
+        FileIO& operator=(FileIO&&)      = delete;
+
+        //Text file read/write
+        std::string ReadTextFile(Directory directory, const std::string& filePath);
+        bool WriteTextFile(Directory directory, const std::string& filePath, const std::string& data);
+
+        //Binary file read/write
+        std::vector<char> ReadBinaryFile(Directory directory, const std::string& filePath);
```

Comment:
Why is the binary file made up of a std::vector<char>? Maybe we could create something like a buffer class that contains the data or pointer from and length to read the data from memory?

Reply:
I am not sure what the best solution is here. But for now I think this is fine.